src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -97,7 +97,7 @@ public function denormalize($data, $type, $format = null, array $context = [])
         $dateTimeFormat = $context[self::FORMAT_KEY] ?? null;
         $timezone = $this->getTimezone($context);
 
-        if ('' === $data || null === $data) {
+        if (null === $data || (\is_string($data) && '' === trim($data))) {
```

Comment:
https://3v4l.org/lgoC9

Reply:
well this changes a lot but exception messages says: `The data is either an empty **string** or **null**, you should pass a **string** that can be parsed with the passed format or a valid DateTime **string**.`